client/gutenberg/extensions/videopress/edit.js

```diff
@@ -0,0 +1,77 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import debugFactory from 'debug';
+import getVideoPressUrl from './get-videopress-url';
+
+const debug = debugFactory( 'extensions:gutenberg:videopress' );
+
+const withVideoPressEdit = createHigherOrderComponent( CoreVideoEdit => {
+	return class VideoPressEdit extends Component {
+		setAttributes = attributes => {
+			const { id, src } = attributes;
+			const { id: currentId } = this.props.attributes;
+			let shouldSetSrcFromVideoPress = false;
+
+			if ( id && id !== currentId ) {
+				shouldSetSrcFromVideoPress = true;
+
+				this.setSrcFromVideoPress( id ).catch( error => {
+					// If fails, fallback to the original src attribute
+					this.props.setAttributes( { src } );
```

Comment:
at this point if we reset to the original `src` then we're introducing a race when the editor made an update during the API call window. just noting here. maybe it's worth tossing in a 

```js
// @TODO: Disable changing the video source while waiting on API call to return to prevent race with successive editor uptakes
```